File: app/src/main/java/uk/co/ourfriendirony/medianotifier/clients/rawg/game/search/GameSearchAddedByStatus.java
package uk.co.ourfriendirony.medianotifier.clients.rawg.game.search;

import com.fasterxml.jackson.annotation.JsonAnyGetter;
import com.fasterxml.jackson.annotation.JsonAnySetter;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;

import java.util.HashMap;
import java.util.Map;

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonPropertyOrder({
        "yet",
        "owned",
        "beaten",
        "toplay",
        "dropped",
        "playing"
})
public class GameSearchAddedByStatus {

    @JsonProperty("yet")
    private Integer yet;
    @JsonProperty("owned")
    private Integer owned;
    @JsonProperty("beaten")
    private Integer beaten;
    @JsonProperty("toplay")
    private Integer toplay;
    @JsonProperty("dropped")
    private Integer dropped;
    @JsonProperty("playing")
    private Integer playing;
    @JsonIgnore
    private Map<String, Object> additionalProperties = new HashMap<String, Object>();

    @JsonProperty("yet")
    public Integer getYet() {
        return yet;
    }

    @JsonProperty("yet")
    public void setYet(Integer yet) {
        this.yet = yet;
    }

    @JsonProperty("owned")
    public Integer getOwned() {
        return owned;
    }

    @JsonProperty("owned")
    public void setOwned(Integer owned) {
        this.owned = owned;
    }

    @JsonProperty("beaten")
    public Integer getBeaten() {
        return beaten;
    }

    @JsonProperty("beaten")
    public void setBeaten(Integer beaten) {
        this.beaten = beaten;
    }

    @JsonProperty("toplay")
    public Integer getToplay() {
        return toplay;
    }

    @JsonProperty("toplay")
    public void setToplay(Integer toplay) {
        this.toplay = toplay;
    }

    @JsonProperty("dropped")
    public Integer getDropped() {
        return dropped;
    }

    @JsonProperty("dropped")
    public void setDropped(Integer dropped) {
        this.dropped = dropped;
    }

    @JsonProperty("playing")
    public Integer getPlaying() {
        return playing;
    }

    @JsonProperty("playing")
    public void setPlaying(Integer playing) {
        this.playing = playing;
    }

    @JsonAnyGetter
    public Map<String, Object> getAdditionalProperties() {
        return this.additionalProperties;
    }

    @JsonAnySetter
    public void setAdditionalProperty(String name, Object value) {
        this.additionalProperties.put(name, value);
    }

}

File: app/src/main/java/uk/co/ourfriendirony/medianotifier/clients/rawg/game/search/GameSearchAddedByStatus.kt
package uk.co.ourfriendirony.medianotifier.clients.rawg.game.search

import com.fasterxml.jackson.annotation.*

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties(ignoreUnknown = true)
@JsonPropertyOrder("yet", "owned", "beaten", "toplay", "dropped", "playing")
class GameSearchAddedByStatus {
    @get:JsonProperty("yet")
    @set:JsonProperty("yet")
    @JsonProperty("yet")
    var yet: Int? = null

    @get:JsonProperty("owned")
    @set:JsonProperty("owned")
    @JsonProperty("owned")
    var owned: Int? = null

    @get:JsonProperty("beaten")
    @set:JsonProperty("beaten")
    @JsonProperty("beaten")
    var beaten: Int? = null

    @get:JsonProperty("toplay")
    @set:JsonProperty("toplay")
    @JsonProperty("toplay")
    var toplay: Int? = null

    @get:JsonProperty("dropped")
    @set:JsonProperty("dropped")
    @JsonProperty("dropped")
    var dropped: Int? = null

    @get:JsonProperty("playing")
    @set:JsonProperty("playing")
    @JsonProperty("playing")
    var playing: Int? = null
}